src/lib.rs

```diff
@@ -1,9 +1,20 @@
 pub mod electrum_extended_priv_key;
 pub mod electrum_extended_pub_key;
+#[cfg(feature = "wallet_file")]
+pub mod electrum_wallet_file;
 
 pub use electrum_extended_priv_key::ElectrumExtendedPrivKey;
 pub use electrum_extended_pub_key::ElectrumExtendedPubKey;
+#[cfg(feature = "wallet_file")]
+pub use electrum_wallet_file::ElectrumWalletFile;
 
 pub trait ElectrumExtendedKey {
+    /// Returns internal and external descriptor
     fn to_descriptors(&self) -> Vec<String>;
+
+    /// Returns the bitcoin extended key (xpub or xprv) as String
+    fn xkeystr(&self) -> String;
```

Comment:
Can't we just `impl Display for ElectrumExtendedKey ` and use `to_string()`?

If there are good reasons to not implement Display, I don't like the fn name, it may refer to x-only keys? Should we add `_` to facilitate reading?

Reply:
Yes that looks cleaner. Iam always a bit hesitant to feed something to Display that is not a 1to1 representation of the internals, but some transformation.

Reply:
Actually, you are right, I am sorry, `impl Display for ElectrumExtendedKey` should print the electrum version key and not lose information.
 
I guess I would expose borrow xprv/xpub and call &exkey.xpub().to_string() when needed

just FYI the impl Display you provided could be done without allocating the string by just calling fmt on the inner element ie `self.xprv.fmt(f)` instead of `write!(f, "{}", self.xprv.to_string())`

Sorry again for the back-and-forth

Reply:
Ok, reverted the Display stuff, and added a _ to the function name to make it xkey_str(). 
Returning a ref doesn't work because of local temporaries.
The reason I added the function to ElectrumExtendedKey is that at the place where I call the function, I don't know if I hold a pub or priv key. The instance comes from get_xkey().